.azure-pipelines/env.yml

```diff
@@ -142,6 +142,59 @@ jobs:
     displayName: Set development/release env
     name: state
 
+  - bash: |
+      # TODO(phlax): move this to a script to ensure proper linting etc
+      set -e
+
+      # Run everything in postsubmit
+      if [[ "$(Build.Reason)" != "PullRequest" ]]; then
+          echo "##vso[task.setvariable variable=build;isoutput=true]true"
+          echo "##vso[task.setvariable variable=checks;isoutput=true]true"
+          echo "##vso[task.setvariable variable=docker;isoutput=true]true"
+          echo "##vso[task.setvariable variable=packaging;isoutput=true]true"
+          exit 0
+      fi
+
+      RUN_BUILD=true
+      RUN_CHECKS=true
+      RUN_DOCKER=true
+      RUN_PACKAGING=true
+      if [[ $(changed.mobileOnly) == true && $(changed.docsOnly) == true ]]; then
+          RUN_BUILD=false
+          RUN_DOCKER=false
+      fi
+      if [[ "$(changed.mobileOnly)" == true && $(changed.docsOnly) == true && $(changed.examplesOnly) == true ]]; then
+          RUN_CHECKS=false
+          RUN_PACKAGING=false
+      fi
```

Comment:
If these `&&` semantics right? I am not familiar to the ci script, so i am not sure.

Reply:
its an unholy mess of azp and bash vars - i believe so ive tested reasonably well

i will factor these out (in a follow up) to a script so we properly separate the azp mess from normal bash scripting

Reply:
as explained offline - the logic is incorrect - updated - thanks @wbpcode - great spot!